eventbus/driver/nats.go

```diff
@@ -260,7 +305,7 @@ func (n *natsStreaming) processEventSourceMsg(m *stan.Msg, msgHolder *eventSourc
 	go action(messages)
 
 	msgHolder.reset(depName)
-	_ = m.Ack()
+	msgHolder.ackAndCache(m, event.ID())
```

Comment:
You are right. However, as I said above, it is expected that 1 message is delivered multiple times because we need to wait for all the dependent event sources to receive messages before acking them... the only differences for these 2 cases is, duplicate delivery that you saw happens almost at the same time, the other one is redelivery happens in `AckWait` seconds - just realized we should set `AckWait` back to 1 second.

I think the way to handle it is adding another check with `loadAndStore` before line 296, and simply ack it at 308. thoughts?

Reply:
gotcha, i think it makes sense to add the load or store just before calling `go action` then ack below

Reply:
And we don't need to make this changer since the function is a blocking call as I mentioned.